### DISCUSSION TITLE
Are tarballs stably versioned?

### DISCUSSION THREAD

**Discussion by mfansler** [r-universe-org/help#262] [Q&A] [answered]
A tarball downloaded from `https://mlr-org.r-universe.dev` on May 2 had a different checksum than now. I filed an issue on the package repository (https://github.com/mlr-org/mlr3proba/issues/327). 

However, I'm wondering if this is really an R-universe issue, since the repository doesn't appear to have made any updates since 27 March 2023, but the files in newer tarball all have creation dates of 26 May 2023. Is this due to an R-universe rebuild of the tarball?

Would it be possible to have checksums available for the hosted files?

**Comment by jeroen** [comment, marked as answer]
Indeed, the packages are continuously rebuilt and rechecked, to keep them compatible with the latest versions of dependencies. Because R stamps a `Packaged` timestamp in the [DESCRIPTION](https://mlr-org.r-universe.dev/paradox/DESCRIPTION) file, the checksum will be different each time.

You can find `MD5sum` of all files in the API: https://mlr-org.r-universe.dev/api/packages
